### ISSUE TITLE
random static files 404

### ISSUE THREAD

**Issue by filmaj** [adobe/developer.adobe.com#133]
latest master of this site, revision 2a8d08c, deployed to https://adobedevsite.helix-demo.xyz/

deployment output:

```
➜ hlx build && hlx deploy --no-auto && hlx publish
✨  Built in 1.35s.

.hlx/build/docs_html.pre.map             15.94 KB     20ms
.hlx/build/html.js                       11.79 KB    715ms
.hlx/build/docs_html.js                  11.54 KB    677ms
.hlx/build/default_html.pre.map          10.97 KB     20ms
.hlx/build/summary_html.pre.map          10.34 KB     19ms
.hlx/build/html.map                       8.84 KB     19ms
.hlx/build/docs_html.pre.js                8.7 KB    508ms
.hlx/build/docs_html.map                  5.87 KB      1ms
.hlx/build/summary_html.pre.js            5.67 KB    543ms
.hlx/build/default_html.pre.js            5.61 KB    445ms
.hlx/build/summary_html.js                4.57 KB    695ms
.hlx/build/DOM_munging.map                4.03 KB     19ms
.hlx/build/sitemap_xml.js                 3.58 KB    748ms
.hlx/build/default_html.js                3.54 KB    605ms
.hlx/build/html.pre.map                   3.12 KB     19ms
.hlx/build/DOM_munging.js                 2.12 KB    282ms
.hlx/build/sitemap_xml.map                2.07 KB     19ms
.hlx/build/mountpoint_resolution.map      1.97 KB     18ms
.hlx/build/html.pre.js                    1.92 KB    303ms
.hlx/build/summary_html.map                1.3 KB     19ms
.hlx/build/mountpoint_resolution.js       1.25 KB    314ms
.hlx/build/default_html.map                 387 B      1ms
.hlx/build/html.pre.type                      2 B    300ms
.hlx/build/docs_html.pre.type                 2 B    506ms
.hlx/build/DOM_munging.type                   2 B    280ms
.hlx/build/summary_html.pre.type              2 B    541ms
.hlx/build/default_html.pre.type              2 B    443ms
.hlx/build/mountpoint_resolution.type         2 B    311ms
[==================================================] analyzing sitemap_xml.js 0.0s
[==================================================] packaging whatwg-encoding 0.0s
✅  packaging completed
[==================================================]  0.0s

✅  deployment of 8 actions completed:
   - developer-adobe-com/c20ea5c5b64b5b7373bf6ff1c759c63678e90709/default_html (8.76 MiB)
   - developer-adobe-com/c20ea5c5b64b5b7373bf6ff1c759c63678e90709/docs_html (7.78 MiB)
   - developer-adobe-com/c20ea5c5b64b5b7373bf6ff1c759c63678e90709/DOM_munging (1.34 KiB)
   - developer-adobe-com/c20ea5c5b64b5b7373bf6ff1c759c63678e90709/html (7.59 MiB)
   - developer-adobe-com/c20ea5c5b64b5b7373bf6ff1c759c63678e90709/sitemap_xml (5.91 MiB)
   - developer-adobe-com/c20ea5c5b64b5b7373bf6ff1c759c63678e90709/sitemap_xml.pre (1.77 KiB)
   - developer-adobe-com/c20ea5c5b64b5b7373bf6ff1c759c63678e90709/static (1.88 MiB)
   - developer-adobe-com/c20ea5c5b64b5b7373bf6ff1c759c63678e90709/summary_html (7.59 MiB)
Affected strains of ssh://git@github.com/adobe/developer.adobe.com.git#master:
- helix-demo
Updated helix-config.yaml
Publishing [==================================================] action root 0.0s



Publishing [==================================================] activating version 0.0s
Publishing [==================================================] activated version 40 0.0s
Publishing [==================================================]  0.0s

✅  The following strains have been published and version 40 is now online:
- default
- launch-docs-helixdemo -> https://adobedevsite.helix-demo.xyz/launch/docs
- xd-docs-helixdemo -> https://adobedevsite.helix-demo.xyz/xd/docs
- starter-docs-helixdemo -> https://adobedevsite.helix-demo.xyz/starter/docs
- helix-demo -> https://adobedevsite.helix-demo.xyz/
- launch-docs-prod -> https://www.adobe.io/launch/docs
- launch-docs-stage -> https://stage.adobe.io/launch/docs
- launch-docs-local -> http://localhost:3000/launch/docs
- xd-docs-prod -> https://www.adobe.io/xd/docs
- xd-docs-stage -> https://stage.adobe.io/xd/docs
- xd-docs-local -> http://localhost:3000/xd/docs
- xd-marketing-prod -> https://www.adobe.io/xd
- xd-marketing-stage -> https://stage.adobe.io/xd
- xd-marketing-local -> https://localhost:3000/xd
- starter-docs-prod -> https://www.adobe.io/starter/docs
- starter-docs-stage -> https://stage.adobe.io/starter/docs
- starter-docs-local -> http://localhost:3000/starter/docs

You now access your site using:
$ curl https://adobedevsite.helix-demo.xyz/launch/docs
```

`hlx` version 0.13.5

**Comment by filmaj** [comment]
I believe @tripodsan is investigating this.

**Comment by tripodsan** [comment]
my current findings are:
- using single requests, I never get a 404.
- using concurrent requests, I sometimes get a 404.

![image](https://user-images.githubusercontent.com/917628/53137209-31b0a380-35c5-11e9-9702-8660f5d6e5a3.png)

It might be related to the bug that @drwilco worked on during the hackathon, such as the requests to the static always go first to the `html` action, IIUC. and if that has a concurrency problem, the static ones might timeout. or if fetching from github then produces a rate-limit problem.

**Comment by tripodsan** [comment]
increasing the concurrency count gives more errors:

![image](https://user-images.githubusercontent.com/917628/53137311-979d2b00-35c5-11e9-9339-200e977a10b5.png)

**Comment by trieloff** [comment]
We might have to move this issue to `helix-cli`.

**Comment by trieloff** [comment]
With https://github.com/adobe/helix-cli/pull/443 merged, we might get a few more insights, as the backend error will be passed through. I doubt it's `raw.githubusercontent.com`, though, as I've been unable to get it to give me anything but 200s - even with 200 concurrent requests.

**Comment by tripodsan** [comment]
I saw that `raw.githubusercontent.com` also uses fastly:

```
$ curl -v https://raw.githubusercontent.com/tripodsan/hlxtest/master/htdocs/dist/favicon.ico > /dev/null
> GET /tripodsan/hlxtest/master/htdocs/dist/favicon.ico HTTP/1.1
> Host: raw.githubusercontent.com
...
< HTTP/1.1 200 OK
...
< X-Fastly-Request-ID: 96eeacacbe4d7c9d47c1b558836ebd61410e2168
<
```

so I think not every request will get through to the backend and might not count against the rate limit.

**Comment by trieloff** [comment]
Yes, they are doing some pretty cool stuff with Fastly (like bulk certificate management)

**Comment by filmaj** [comment]
I tried to deploy the site out w/ latest helix-cli (`0.13.6-pre.6`) in order to get the latest logging out there but I get an error when I try to `hlx build`. I tried deleting both the `.cache` and `.hlx` directories but that didn't help.

```
Invalid configuration:
Configuration  should be object: type([{"version":1,"definitions":{"defaults":["http://localhost/local/default.git"],"strains":{"base":{"code":"http://localhost/local/default.git","content":"http://localhost/local/default.git","static":"http://localhost/local/default.git","package":"16cd39e4810356ecc384bb5185f081e23cae229b"},"proxy":{"origin":{"address":"34.227.208.69","use_ssl":true,"ssl_cert_hostname":"www.adobe.io"}}}},"strains":{"default":{"code":"http://localhost/local/default.git","content":"http://localhost/local/default.git","static":"http://localhost/local/default.git","package":"9e81a2d5d82f2a22bf218b446da32414f33c0d87"},"launch-docs-helixdemo":{"code":"http://localhost/local/default.git","content":"https://github.com/Adobe-Marketing-Cloud/reactor-user-docs.git#master","static":"http://localhost/local/default.git","package":"77017e262a9aa0378abd02c098ecc82e9b6aa7dc","url":"https://adobedevsite.helix-demo.xyz/launch/docs","directoryIndex":"README.html"},"xd-docs-helixdemo":{"code":"http://localhost/local/default.git","content":"https://github.com/AdobeXD/plugin-docs.git#master","static":"http://localhost/local/default.git","package":"77017e262a9aa0378abd02c098ecc82e9b6aa7dc","url":"https://adobedevsite.helix-demo.xyz/xd/docs","directoryIndex":"README.html"},"starter-docs-helixdemo":{"code":"http://localhost/local/default.git","content":"https://github.com/AdobeDocs/starter-docs.git#master","static":"http://localhost/local/default.git","package":"77017e262a9aa0378abd02c098ecc82e9b6aa7dc","url":"https://adobedevsite.helix-demo.xyz/starter/docs","directoryIndex":"Overview.html"},"helix-demo":{"url":"https://adobedevsite.helix-demo.xyz/","code":"ssh://git@github.com/adobe/developer.adobe.com.git#master","content":"ssh://git@github.com/adobe/developer.adobe.com.git#master","static":"https://github.com/adobe/developer.adobe.com.git/htdocs#master","package":"developer-adobe-com/c20ea5c5b64b5b7373bf6ff1c759c63678e90709"}}},{"launch-docs-local":null},null,{"code":"http://localhost/local/default.git","content":"http://localhost/local/default.git","static":"http://localhost/local/default.git","package":"16cd39e4810356ecc384bb5185f081e23cae229b"},{"url":"http://localhost:3000/launch/docs","content":"https://github.com/Adobe-Marketing-Cloud/reactor-user-docs.git#master","directoryIndex":"README.html","package":"9e81a2d5d82f2a22bf218b446da32414f33c0d87"},{"xd-docs-local":{"code":"http://localhost/local/default.git","content":"https://github.com/AdobeXD/plugin-docs.git#master","static":"http://localhost/local/default.git","package":"9e81a2d5d82f2a22bf218b446da32414f33c0d87","url":"http://localhost:3000/xd/docs","directoryIndex":"README.html"}},{"xd-marketing-local":{"code":"http://localhost/local/default.git","content":"https://github.com/AdobeXD/plugin-docs.git#author","static":"http://localhost/local/default.git","package":"9e81a2d5d82f2a22bf218b446da32414f33c0d87","url":"https://localhost:3000/xd","directoryIndex":"README.html"}},{"starter-docs-local":{"code":"http://localhost/local/default.git","content":"https://github.com/AdobeDocs/starter-docs.git#author","static":"http://localhost/local/default.git","package":"9e81a2d5d82f2a22bf218b446da32414f33c0d87","url":"http://localhost:3000/starter/docs","directoryIndex":"Overview.html"}}], {"type":"object"})

data should be object
```

**Comment by filmaj** [comment]
Nevermind on my last comment! Apparently _how_ you drop comments in yaml matters (see https://github.com/adobe/developer.adobe.com/commit/3624692853fa66cd907cbbdc3b7268adbc1ec02c).

**Comment by filmaj** [comment]
So I deployed w/ latest helix-cli (master) to get the new logging y'all mentioned, loaded the deployed site (https://adobedevsite.helix-demo.xyz/) and I did a bit of digging into the activation logs.

First, here's the network tab from loading the deployed site in chrome:

<img width="1414" alt="screen shot 2019-02-22 at 4 44 42 pm" src="https://user-images.githubusercontent.com/52645/53275593-3e1a2680-36c1-11e9-86a1-0284788a71ae.png">

I then painstakingly sifted through the `wsk activation list` to see what I could find from the failing static file requests.

Here is the activation details for the `spectrum-darkest.css` file:

```
➜ wsk activation get f2e54cda6d3c459aa54cda6d3c459a11
ok: got activation f2e54cda6d3c459aa54cda6d3c459a11
{
    "namespace": "developer-adobe-com",
    "name": "hlx--static",
    "version": "0.0.2",
    "subject": "developer-adobe-com",
    "activationId": "f2e54cda6d3c459aa54cda6d3c459a11",
    "start": 1550874866197,
    "end": 1550874866252,
    "duration": 55,
    "response": {
        "status": "success",
        "statusCode": 0,
        "success": true,
        "result": {
            "body": "/* Temporary skin variables... blah blah blah lots of correct, expected spectrum-css content here",
            "headers": {
                "Cache-Control": "s-max-age=300",
                "Content-Type": "text/css",
                "ETag": "\"/5zMlZLaZwhMSPwPpPgkKA==\"",
                "X-Static": "Raw/Static"
            }
        }
    },
    "logs": [],
    "annotations": [
        {
            "key": "limits",
            "value": {
                "concurrency": 1,
                "logs": 10,
                "memory": 256,
                "timeout": 60000
            }
        },
        {
            "key": "path",
            "value": "developer-adobe-com/hlx--static"
        },
        {
            "key": "kind",
            "value": "nodejs:10-fat"
        },
        {
            "key": "waitTime",
            "value": 6
        }
    ],
    "publish": false
}
```

So, the activation details seem to suggest that the correct response is being sent from runtime, right?

The response headers returned to my browser were as follows:

```
accept-ranges: bytes
age: 0
content-length: 4
content-type: application/json
date: Fri, 22 Feb 2019 22:34:48 GMT
status: 404
strict-transport-security: max-age=31536000; includeSubDomains
vary: X-Debug, X-Strain
x-status: 404-Restart 1
x-version: 41; src=41; cli=0.13.6-pre.6; rev=3624692853fa66cd907cbbdc3b7268adbc1ec02c
```

Not sure what the `x-status` header is saying, but I decided to post this just in case.

Finally, the logs for this activation are:

```
➜ wsk activation logs f2e54cda6d3c459aa54cda6d3c459a11
2019-02-22T22:34:26.199Z       stdout: main() adobe developer.adobe.com master /spectrum-darkest.css /spectrum-darkest.css helix-demo true   /htdocs
2019-02-22T22:34:26.199Z       stdout: deliverPlain() adobe developer.adobe.com master htdocs/spectrum-darkest.css
2019-02-22T22:34:26.248Z       stdout: size 136280
2019-02-22T22:34:26.248Z       stdout: delivering file htdocs/spectrum-darkest.css type text/css binary: false
```

I was also able to find successful activations for all of the other 404'd files like `spectrum-core.css`, `spectrum-flexbox-grid.css`, `spectrum-light.css`,  `spectrum-lightest.css`, `spectrum-darkest.css`, `spectrum-dark.css`, `footer.css`.

This leads me to believe this problem is not due to anything happening on runtime. I unfortunately do not know how to debug stuff at the fastly level.

**Comment by tripodsan** [comment]
nice finding! 

what is weird is the statuscode:

```
        "statusCode": 0,
```

I don't know what runtime will send back to fastly, but I guess not a 200 :-(
so maybe this is the problem? I'll take a closer look again next week.

**Comment by filmaj** [comment]
I tried redeploying the site with the latest helix-cli master, which includes the fix from https://github.com/adobe/helix-cli/issues/585 (always return status code 200) but that didn't seem to help much. 😞

**Comment by filmaj** [comment]
FWIW even after deploying w/ helix-cli `0.13.6-pre.8`, I still see the `statusCode` from the runtime response (as per `wsk activation get <id>`) saying statusCode: 0.

**Comment by filmaj** [comment]
Per @rofe, it seems like the packaged `static.zip` used for the static action was not updated as part of me using the latest helix-cli. I've had to modify the deploy script so that it wipes out the `.hlx` directory before deploying (see 9b235484fc3276b3c663f9aabca93d9f07d15c4c). Now can confirm the latest static action has the correct code, including returning the proper status code... tho only in one area. E.g.:

```
➜ wsk activation get be6e9d76742640ccae9d76742650cc57
ok: got activation be6e9d76742640ccae9d76742650cc57
{
    "namespace": "developer-adobe-com",
    "name": "hlx--static",
    "version": "0.0.2",
    "subject": "developer-adobe-com",
    "activationId": "be6e9d76742640ccae9d76742650cc57",
    "start": 1551132343062,
    "end": 1551132343109,
    "duration": 47,
    "response": {
        "status": "success",
        "statusCode": 0, <======== still zero
        "success": true,
        "result": {
            "body": "blahblahblah",
            "headers": {
                "Cache-Control": "s-max-age=300",
                "Content-Type": "image/vnd.microsoft.icon",
                "ETag": "\"JPVrmjI8kr96VCjLtnZeqw==\"",
                "X-Static": "Raw/Static"
            },
            "statusCode": 200 <==== oh hey
        }
    },
    "logs": [],
    "annotations": [
        {
            "key": "limits",
            "value": {
                "concurrency": 1,
                "logs": 10,
                "memory": 256,
                "timeout": 60000
            }
        },
        {
            "key": "path",
            "value": "developer-adobe-com/hlx--static"
        },
        {
            "key": "kind",
            "value": "nodejs:10-fat"
        },
        {
            "key": "waitTime",
            "value": 6
        }
    ],
    "publish": false
}
```

**Comment by filmaj** [comment]
I've noticed something else that seems weird to me. Not sure if worth spinning out into a separate issue, or if it is related to the static file failures or not, but I will dump a report in here anyways. We can spin it out into different issues as we see fit.

So, as has been documented in this issue, loading the devsite usually yields a bunch of 404 responses to static resources such as CSS, sometimes JS, sometimes SVG files. After a most recent deploy, when I went to load the site I no 404s - 200s across the board (yay! I thought this was maybe related to the recent change in ensuring a 200 status code was being returned in static handlers). If I then run `wsk activation list`, here's what I see (a total of 26 runtime action invocations):

```
➜ wsk activation list --since 1551132033612 -l 50
activations
be6e9d76742640ccae9d76742650cc57 hlx--static
88ec23a125044c99ac23a12504ec994d html
c09bffb2257249d59bffb22572b9d5e6 hlx--static
2acd6c137d2f4b498d6c137d2f6b49de html
a39653430adb4ae19653430adb6ae187 hlx--static
5952b948622b4a5292b948622b8a5238 html
fb5b0fae4bc94cfc9b0fae4bc95cfc13 hlx--static
65c7037f8e68474987037f8e682749e0 hlx--static
227d151845634a7abd151845633a7a50 hlx--static
0bd8a0a7530044ad98a0a75300c4ad5a hlx--static
11d3196725a64eef93196725a6eeefc1 hlx--static
7208ef56a965432788ef56a965a327ef hlx--static
89c15d7deefd48dd815d7deefde8ddff hlx--static
d6269a6188be4f70a69a6188be5f703d hlx--static
ae9515e82394470d9515e82394470dca hlx--static
d3fda9c2afd34974bda9c2afd3b97484 hlx--static
69a6d5e55d8341f3a6d5e55d8301f320 html
6fc6dcd51bc94d7d86dcd51bc9ed7d35 html
95ead8f8a15843bcaad8f8a15813bc3f html
28a1b41dc659488ba1b41dc659f88b5d html
d25356ebd41041399356ebd410d13982 html
ecaa5ed345ef4ca5aa5ed345efdca5d9 html
87fd6fba27044f3cbd6fba2704cf3ceb html
85e7704309ce4968a7704309cea9681e html
c3944c8d225c4d02944c8d225c2d02e2 html
1a80289992904681802899929096811b html
```

However, on second load I ended up with 4 stylesheet 404s, and loading the homepage this time yielded 18 runtime activations:

```
➜ wsk activation list --since 1551132343109 -l 50
activations
6cf80ec5c3d64463b80ec5c3d64463ac hlx--static
e61adc0489d849399adc0489d89939da html
f1ed23ab2bde454fad23ab2bde054f20 hlx--static
fa1b4fcff40b471c9b4fcff40bc71c11 hlx--static
406e944cd2cf4fbdae944cd2cfafbdb0 html
0cbc494b1d3145a7bc494b1d3195a7d5 html
ac53bf9fd4e44c2993bf9fd4e4ec2948 hlx--static
8700497a81394ce580497a81397ce519 hlx--static
06f9b05e71dc4463b9b05e71dc84633c hlx--static
20871dc725ac4017871dc725ace0178b html
4a54f90118f3477794f90118f3d777da html
65ed18e48c5a48f0ad18e48c5a88f028 hlx--static
e3fa5040da3d404bba5040da3dd04b6e hlx--static
343b1fe8ff3342c8bb1fe8ff33e2c843 hlx--static
50ffc647e3a541f1bfc647e3a501f156 html
049980bf2e1d406f9980bf2e1dd06f06 html
0d38e62576364f14b8e62576367f145c html
90c590d18bfd42868590d18bfd028684 html
```

First question is, with the way we have the site setup, the `html` handler should be offloading to the `default_html` handler via ESI include - but why don't I see `default_html` being activated in these cases?

Second question is, why is `html` being triggered so often?

If I look at all the activations for the `html` action, they _all_ look like this, regardless of in the no-404s or some-404s case:

```
{
    "namespace": "developer-adobe-com",
    "name": "html",
    "version": "0.0.1",
    "subject": "developer-adobe-com",
    "activationId": "1a80289992904681802899929096811b",
    "start": 1551132336151,
    "end": 1551132336303,
    "duration": 152,
    "response": {
        "status": "success",
        "statusCode": 0,
        "success": true,
        "result": {
            "body": {},
            "headers": {
                "Content-Type": "application/json"
            },
            "statusCode": 404
        }
    },
    "logs": [],
    "annotations": [
        {
            "key": "limits",
            "value": {
                "concurrency": 1,
                "logs": 10,
                "memory": 256,
                "timeout": 60000
            }
        },
        {
            "key": "path",
            "value": "developer-adobe-com/9b235484fc3276b3c663f9aabca93d9f07d15c4c/html"
        },
        {
            "key": "kind",
            "value": "nodejs:10-fat"
        },
        {
            "key": "waitTime",
            "value": 388
        }
    ],
    "publish": false
}
```

They all say `"success": true`, `"Content-Type": "application/json"` and `"statusCode": 404`.

The other thing I keep noticing is the annotations on the runtime actions, particularly the `limits` section. It says `concurrency: 1`. Given that typical browser load many static assets in parallel, is this a problem?

I am just brainstorming and trying to gather as many logs as I can. I still don't have visibility into what is happening on Fastly's end, which is obviously a huge missing piece to this puzzle. However, without solving this issue, we cannot ship the site to production. This is the most critical issue for this site in my mind.

**Comment by tripodsan** [comment]
> Second question is, why is `html` being triggered so often?

because this is how the static delivery currently works.
fastly doesn't know if a (non image) resource is part of the content or part of the static. so it first sends the request to the `html` action. if that returns with a 404, it then sends it to the static action. for images, it checks first the content git repo, and if 404, it asks static.

I see some solutions to improve this:

0. add `.css` and `.js` to be treated by default as static files
1. inverse the logic - i.e. let fastly know for which extensions/selectors that actions exists, and call static for all others.
2. storing a list of all static files in some _mainfest_, include it in an edge dictionary, and let fastly decide very early, if it needs to call static or not.

**Comment by filmaj** [comment]
Alright so I'm not sure what happened but I think this is fixed.

@stevengill deployed the site based on the code in #147, which is just `helix-config.yaml` changes. Now when I go to load https://adobedevsite.helix-demo.xyz/, I don't see any static site failures. I just tried loading the site 20 times in a row in my browser and can't reproduce.

As a result, I am closing this!

**Comment by stevengill** [comment]
Reopening this as the random missing static files is back. We changed nothing from the previous message fil posted. 

Ex: go to https://adobedevsite.helix-demo.xyz/starter/docs

In console the random missing static files change with each refresh. 

```
docs:185 GET https://adobedevsite.helix-demo.xyz/starter/assets/adobe_standard_logo.png?width=3192&auto=webp 404
docs:31 GET https://adobedevsite.helix-demo.xyz/footer.css net::ERR_ABORTED 404
docs:30 GET https://adobedevsite.helix-demo.xyz/topnav.css net::ERR_ABORTED 404
```

**Comment by filmaj** [comment]
Using a chrome extension that @tripodsan recommended to me, I set `X-Debug:true` header for all requests in a browser session.

Interestingly, now for many requests, I get `cancelled` in the chrome network tab. Unclear what this means... need to dig into this more.

<img width="844" alt="screen shot 2019-02-26 at 5 49 25 pm" src="https://user-images.githubusercontent.com/52645/53455053-e516ff00-39ee-11e9-8fe9-644843da1b8e.png">

Note that I still get some 200s.

Finally, inspecting the response headers on the one 404 failure, here is what I see:

```
accept-ranges: bytes
access-control-allow-headers: Authorization, Origin, X-Requested-With, Content-Type, Accept, User-Agent
access-control-allow-methods: OPTIONS, GET, DELETE, POST, PUT, HEAD, PATCH
access-control-allow-origin: *
age: 0
content-length: 4
content-type: application/json
date: Tue, 26 Feb 2019 23:43:11 GMT
server: api-gateway/1.9.3.1
status: 404
strict-transport-security: max-age=31536000; includeSubDomains
vary: X-Debug, X-Strain
via: 1.1 varnish
x-action-root: /developer-adobe-com/c45cb8ff2613aae195c8a5ecd97b8d2da0902413/
x-backend-name: 0gDEZCgYFsF773b88iAgvS--F_AdobeRuntime
x-backend-url: /api/v1/web/developer-adobe-com/c45cb8ff2613aae195c8a5ecd97b8d2da0902413/html?owner=adobe&repo=developer.adobe.com&ref=master&path=/api/v1/web/developer-adobe-com/c45cb8ff2613aae195c8a5ecd97b8d2da0902413/index.md&selector=&extension=&strain=helix-demo&rootPath=&params=(null)
x-cache: HIT
x-cache-hits: 4
x-cdn-request-id: 6e1bb61b-30e4-4099-9a64-6571ff008c79
x-content-type-options: nosniff
x-frame-options: deny
x-fulldirname: /api/v1/web/developer-adobe-com/c45cb8ff2613aae195c8a5ecd97b8d2da0902413/js
x-github-static-ref: @(null)
x-index: index.html
x-openwhisk-activation-id: 3af4a93abcd94cdab4a93abcd97cda96
x-repo-root-path: 
x-request-id: 0CSHdkJqDYpEsTPY81oHAnagqsx8mwOR
x-served-by: cache-dfw18648-DFW
x-static: Static
x-status: 404-Restart 1
x-strain: helix-demo
x-timer: S1551224591.318033,VS0,VE227
x-trace: vcl_recv; hlx_headers_recv; hlx_backend_recv; hlx_strain; hlx_block_recv; hlx_deny; hlx_allow; hlx_recv_pipeline; hlx_owner; hlx_repo; hlx_ref; hlx_root_path; hlx_action_root; vcl_recv; hlx_headers_recv; hlx_backend_recv; hlx_strain; hlx_block_recv; hlx_deny; hlx_allow; hlx_recv_pipeline; hlx_owner; hlx_repo; hlx_ref; hlx_root_path; hlx_index; hlx_action_root; vcl_hit; vcl_deliver; hlx_headers_deliver
x-url: /icons/loadIcons.js
x-version: 47; src=47; cli=0.13.6-pre.8; rev=dcdd7f967604e00f1c1882d4cf8727476f720288
x-xss-protection: 1; mode=block
```

**Comment by tripodsan** [comment]
I modified the (live) VCL to add more information. this is the trace of a good response:

```
 x-trace: vcl_recv; hlx_headers_recv; hlx_backend_recv; hlx_strain; hlx_block_recv; hlx_deny; hlx_allow; hlx_recv_pipeline; hlx_owner; hlx_repo; hlx_ref; hlx_root_path; hlx_action_root; /developer-adobe-com/c45cb8ff2613aae195c8a5ecd97b8d2da0902413/css; END_RECV; vcl_recv; hlx_headers_recv; hlx_backend_recv; hlx_strain; hlx_block_recv; hlx_deny; hlx_allow; hlx_recv_pipeline; hlx_owner; hlx_repo; hlx_ref; hlx_root_path; hlx_index; hlx_action_root; /developer-adobe-com/c45cb8ff2613aae195c8a5ecd97b8d2da0902413/html; END_RECV; vcl_miss; vcl_fetch; hlx_fetch_errors; hlx_headers_fetch; RESTART; vcl_recv; hlx_headers_recv; hlx_backend_recv; hlx_strain; hlx_block_recv; hlx_deny; hlx_allow; hlx_recv_static; hlx_github_static_owner; hlx_github_static_repo; hlx_github_static_ref; hlx_github_static_root; END_RECV; vcl_miss; vcl_deliver; hlx_headers_deliver
```

this one of a bad response:

```
x-trace: vcl_recv; hlx_headers_recv; hlx_backend_recv; hlx_strain; hlx_block_recv; hlx_deny; hlx_allow; hlx_recv_pipeline; hlx_owner; hlx_repo; hlx_ref; hlx_root_path; hlx_action_root; /developer-adobe-com/c45cb8ff2613aae195c8a5ecd97b8d2da0902413/js; END_RECV; vcl_recv; hlx_headers_recv; hlx_backend_recv; hlx_strain; hlx_block_recv; hlx_deny; hlx_allow; hlx_recv_pipeline; hlx_owner; hlx_repo; hlx_ref; hlx_root_path; hlx_index; hlx_action_root; /developer-adobe-com/c45cb8ff2613aae195c8a5ecd97b8d2da0902413/html; END_RECV; vcl_hit; vcl_deliver; hlx_headers_deliver
```

**Comment by tripodsan** [comment]
also note, for the 2nd backend request, the `path` parameter is wrong....

**Comment by tripodsan** [comment]
I disabled caching of 404 completely in `vlc_fetch` and now it seems to work better:


```
  } elseif ((beresp.status == 404 || beresp.status == 204) && !req.http.X-Disable-Static) {
    # That was a miss. Let's try to restart.
    set beresp.http.X-Status = beresp.status + "-Restart " + req.restarts;
    set beresp.status = 404;
    # ensure that 404 from the backends are not cached. otherwise the a potential vlc_hit will not re-fetch the static files.
    set beresp.cacheable = false;    
...
```

**Comment by tripodsan** [comment]
btw, I fixed it directly in the active fastly VCL. so don't re-publish before https://github.com/adobe/helix-cli/issues/591 is fixed :-)

**Comment by filmaj** [comment]
Yeahhhhh! HUGE!
